dev-libs/libsecp256k1/libsecp256k1-0.2.0-r2.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools multilib-minimal
+
+MyPN=secp256k1
+DESCRIPTION="Optimized C library for EC operations on curve secp256k1"
+HOMEPAGE="https://github.com/bitcoin-core/secp256k1"
+SRC_URI="
+	https://github.com/bitcoin-core/secp256k1/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/bitcoin-core/secp256k1/commit/772e747bd9104d80fe531bed61f23f75342d7d63.patch -> ${PN}-PR1159-772e74.patch
+"
+
+LICENSE="MIT"
+SLOT="0/1"  # subslot is "$((_LIB_VERSION_CURRENT-_LIB_VERSION_AGE))" from configure.ac
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="+asm +ecdh experimental +extrakeys lowmem +recovery +schnorr static-libs test valgrind"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	asm? ( || ( amd64 arm ) arm? ( experimental ) )
+	schnorr? ( extrakeys )
+"
+
+BDEPEND="
+	sys-devel/autoconf-archive
+	virtual/pkgconfig
+	valgrind? ( dev-util/valgrind )
+"
+
+PATCHES=(
+	"${DISTDIR}/${PN}-PR1159-772e74.patch"
+	"${FILESDIR}/0.2.0-fix-out-of-source-tree-build.patch"
+)
+
+S="${WORKDIR}/${MyPN}-${PV}"
+
+src_unpack() {
+	unpack "${P}.tar.gz"
+}
+
+src_prepare() {
+	default
+	eautoreconf
+
+	# Generate during build
+	rm -f src/precomputed_ecmult.c src/precomputed_ecmult_gen.c || die
+}
+
+multilib_src_configure() {
+	local myconf=(
+		$(use_enable static{-libs,})
```

Comment:
Do we need USE=static-libs? We tend to only want to add it if a package actually requires it.

(https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0302)

Reply:
Is the assumption that the only consumers of a package live within the same repository? In other words, we can't possibly know all the dependents of this package. I generally try to expose all useful `configure` options as `USE` flags. I personally know of no consumers of `libsecp256k1.a`, but I didn't want to assume anyone's use case. I'll drop this addition at your request, though.

Reply:
I would be open to it if we know of a solid use outside of ::gentoo that fits within the kind of case listed on the policy site, but I wouldn't want to add it pre-emptively. Thanks.